routes/artifacts.js

```diff
@@ -0,0 +1,597 @@
+var api     = require('./v1');
+var debug   = require('debug')('queue:routes:resources');
+var _       = require('lodash');
+var assert  = require('assert');
+var Promise = require('promise');
+
+// Common schema prefix
+const SCHEMA_PREFIX_CONST = 'http://schemas.taskcluster.net/queue/v1/';
+
+// Maximum number of artifacts to list
+const MAX_ARTIFACTS_LISTING = 10 * 1000;
+
+/** Post artifact */
+api.declare({
+  method:     'post',
+  route:      '/task/:taskId/runs/:runId/artifacts/:name(*)',
+  name:       'createArtifact',
+  scopes: [
+    [
+      'queue:create-artifact:<name>',
+      'assume:worker-id:<workerGroup>/<workerId>'
+    ]
+  ],
+  deferAuth:  true,
+  input:      SCHEMA_PREFIX_CONST + 'post-artifact-request.json#',
+  output:     SCHEMA_PREFIX_CONST + 'post-artifact-response.json#',
+  title:      "Create Artifact",
+  description: [
+    "This API end-point creates an artifact for a specific run of a task. This",
+    "should **only** be used by a worker currently operating on this task, or",
+    "from a process running within the task (ie. on the worker).",
+    "",
+    "All artifacts must specify when they `expires`, the queue will",
+    "automatically take care of deleting artifacts past their",
+    "expiration point. This features makes it feasible to upload large",
+    "intermediate artifacts from data processing applications, as the",
+    "artifacts can be set to expire a few days later.",
+    "",
+    "We currently support 4 different `storageType`s, each storage type have",
+    "slightly different features and in some cases difference semantics.",
+    "",
+    "**S3 artifacts**, is useful for static files which will be stored on S3.",
+    "When creating an S3 artifact is create the queue will return a pre-signed",
```

Comment:
This sentence doesn't really parse...  "When creating an S3 artifact the queue will return..."?